### ISSUE TITLE
[discussion] JSHint can validate .json; JSON Schema written in Java

### ISSUE THREAD

**Issue by paulvi** [angelozerr/eclipse-wtp-json#7]
JSHint can validate .json

**Comment by angelozerr** [comment]
I don't want to use JSHint for validate json, because it requires a JavaScript engine (like Rhino). 

My idea is to validate at first JSON by checking  missed bracket (WTP XML validator uses tokens created by XMLTokenizer (jflex) to detect missed tag close, my idea is to do the same thing than XML).

In the second step, I would like to validate with JSON Schema too. My idea is to provide an extension point to use any JSON Schema Validator (I have not studied that).

**Comment by paulvi** [comment]
What I mean is that both JSHint and ESLint are capable of validating .json

So this project maybe related to Linters that you also work on.

And aren't you considering running Linters using Node.js? Rhino is definitely retired.

**Comment by angelozerr** [comment]
> What I mean is that both JSHint and ESLint are capable of validating .json

Yes but I prefer implementing JSON validation by using SSE region (like XML WTP)

> So this project maybe related to Linters that you also work on.

As I said you, I prefer doing like WTP XML.

More I'm not sure that ESLInt and JSHint can validate JSON Schema?

> And aren't you considering running Linters using Node.js? Rhino is definitely retired.

In tern.java, linter (used to validate JS files) https://github.com/angelozerr/tern.java/wiki/Tern-Linter are exectued with node.js by using tern plugins an dnot by Rhino (liek JSHInt Eclipse).

**Comment by paulvi** [comment]
> More I'm not sure that ESLInt and JSHint can validate JSON Schema?

I did quick googling and found some interesting sites, but not references about using JSON Schema with ESLInt and JSHint 

Some links
- https://github.com/natesilva/jayschema
- https://www.jetbrains.com/webstorm/help/using-javascript-code-quality-tools.html
- https://jslinterrors.com

**Comment by angelozerr** [comment]
I would like to use a JSON Schema written in Java.

**Comment by paulvi** [comment]
Java tools for JS is very small part of web and JS universe.
And time shows those solution that work in more case are more widely adopted.

i.e. those schemas may change over time and pure JS tools may be more robust because of having more users.

But possibly if you can automatically generate Java POJO from JSON Schema with tools like https://github.com/joelittlejohn/jsonschema2pojo, then updating will be simple "drop in new JSON Schema"

**Comment by paulvi** [comment]
http://json-schema.org/implementations.html lists some Java libs
- https://github.com/fge/json-schema-validator

**Comment by angelozerr** [comment]
> https://github.com/fge/json-schema-validator

Yes it was my idea to try it. It seems that https://github.com/sabina-jung/JSON-Schema-Validation-Eclipse/blob/master/src/sj/jsonschemavalidation/builder/ValidateJson.java use it too.

But my idea is to manage JSON Schema Processor with extension point in order implement as you wish the JSON Schema Validation.

**Comment by paulvi** [comment]
That is nice finding for me, but now it is seems to late to adopt. 
The same I wish it was published.

@sabina-jung seems not responding to any issues and PRs by @PascalBru @jcompagner 

JSON-Schema-Validation-Eclipse should be added to list too #5

**Comment by jcompagner** [comment]
if you guys have nice patches on https://github.com/Servoy/JSON-Schema-Validation-Eclipse then then i guess we can have our git repo as more leading.

**Comment by angelozerr** [comment]
@jcompagner

My idea is to study how to you have used https://github.com/fge/json-schema-validator inside Eclipse. But I will not use your code. I will do teh same thing than WTP XML Schema (provide a catalog with JSON Schema, etc) and implement the validator with WTP.
